pennylane/qnodes/base.py

```diff
@@ -234,6 +235,51 @@ def print_applied(self):
             else:
                 print("{}({}(wires={}))".format(return_type, op.name, op.wires))
 
+    def draw(self, charset="unicode", show_variable_names=False):
+        """Draw the QNode as a circuit diagram.
+
+        Consider the following circuit as an example:
+
+        .. code-block:: python
+            @qml.qnode(dev)
+            def qfunc(a, w):
+                qml.Hadamard(0)
+                qml.CRX(a, wires=[0, 1])
+                qml.Rot(w[0], w[1], w[2], wires=[1])
+                qml.CRX(-a, wires=[0, 1])
+
+                return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        We can draw the circuit after it has been executed:
+
+        .. code-block:: python
+            >>> result = qfunc(2.3, [1.2, 3.2, 0.7])
+            >>> print(qfunc.draw())
+            0: ──H──╭C────────────────────────────╭C─────────╭┤ ⟨Z ⊗ Z⟩
+            1: ─────╰RX(2.3)──Rot(1.2, 3.2, 0.7)──╰RX(-2.3)──╰┤ ⟨Z ⊗ Z⟩
+            >>> print(qfunc.draw(charset="ascii"))
+            0: --H--+C----------------------------+C---------+| <Z @ Z>
+            1: -----+RX(2.3)--Rot(1.2, 3.2, 0.7)--+RX(-2.3)--+| <Z @ Z>
+            >>> print(qfunc.draw(show_variable_names=True))
+            0: ──H──╭C─────────────────────────────╭C─────────╭┤ ⟨Z ⊗ Z⟩
+            1: ─────╰RX(a)──Rot(w[0], w[1], w[2])──╰RX(-1*a)──╰┤ ⟨Z ⊗ Z⟩
+
+        Args:
+            charset (str, optional): The charset that should be used. Currently, "unicode" and "ascii" are supported.
+            show_variable_names (bool, optional): Show variable names instead of values.
+
+        Raises:
+            ValueError: If the given charset is not supported
+            pennylane.QuantumFunctionError: Drawing is impossible because the underlying CircuitGraph has not yet been constructed
+
+        Returns:
+            str: The circuit representation of the QNode
+        """
+        if self.circuit:
+            return self.circuit.draw(charset=charset, show_variable_names=show_variable_names)
+
+        raise QuantumFunctionError("The QNode can only be drawn after its CircuitGraph has been constructed.")
```

Comment:
```suggestion
        raise RuntimeError("The QNode can only be drawn after its CircuitGraph has been constructed.")
```

Reply:
`QuantumFunctionError` indicates that there's something wrong with the quantum function itself, so a `RuntimeError` might be better. Too bad there's no `UserError` :)